drivers/i2c/i2c_ll_stm32.c

```diff
@@ -189,31 +189,17 @@ static int i2c_stm32_init(const struct device *dev)
 #endif
 
 	if (cfg->pinctrl_list_size != 0) {
```

Comment:
In fact, the `0` size check is not needed, function should just return if given an empty array of pins.